libs/community/langchain_community/document_transformers/doctran_text_translate.py

```diff
@@ -65,3 +116,19 @@ def transform_documents(
             Document(page_content=doc.transformed_content, metadata=doc.metadata)
             for doc in doctran_docs
         ]
+
+    async def _to_thread(self, func, /, *args, **kwargs):
+        """Asynchronously run function *func* in a separate thread.
+
+        Any *args and **kwargs supplied for this function are directly passed
+        to *func*. Also, the current :class:`contextvars.Context` is propagated,
+        allowing context variables from the main thread to be accessed in the
+        separate thread.
+
+        Return a coroutine that can be awaited to get the eventual result of *func*.
+        """
+
+        loop = asyncio.get_event_loop()
```

Comment:
Would using our built in functionality work here, so we don't end up with implementations per document transformer?

https://github.com/langchain-ai/langchain/blob/2aa2bdfa567baac18b4f41abf9e4b4712421b850/libs/core/langchain_core/runnables/config.py#L528-L528

```
return await run_in_execuctor(None, func, *args, **kwargs)
```
instead of _to_thread?

Reply:
Hi @eyurtsev, thanks! That makes a lot of sense. I couldn't find that implementation before. I refactored the code and it's cleaner now.